boot/espressif/main.c

```diff
@@ -61,11 +66,40 @@ int main()
     }
 #endif
 
+#ifdef CONFIG_SECURE_FLASH_ENC_ENABLED
+    int rc;
+
+    /* When the native (not MCUboot's) flash encryption feature is enabled
+     * and it is the first boot (flash still not encrypted), the following
+     * steps are done:
+     *   1) Generate flash encryption EFUSE key
+     *   2) Encrypt flash contents
+     *   3) Burn EFUSE to enable flash encryption on the device
+     */
+    BOOT_LOG_INF("Checking flash encryption...");
+    bool flash_encryption_enabled = esp_flash_encryption_enabled();
+    rc = esp_flash_encrypt_check_and_update();
+    if (rc != ESP_OK) {
+        BOOT_LOG_ERR("Flash encryption check failed (%d).", rc);
+        FIH_PANIC;
+    }
+
+    if (!flash_encryption_enabled && esp_flash_encryption_enabled()) {
+        /* Flash encryption was just enabled for the first time,
+           so issue a system reset to ensure flash encryption
+           cache resets properly */
+        BOOT_LOG_INF("Resetting with flash encryption enabled...");
+        bootloader_reset();
+    }
+    BOOT_LOG_INF("Flash encryption: %d", flash_encryption_enabled);
+#endif
```

Comment:
The Flash Encryption (**FE**) enabling process cannot take place at the same time Secure Boot V2 (**SBv2**) is being enabled.

In such scenario, SBv2 would already have enqueued some eFuse operations before FE process begins. In case the FE succeeds, all the pending eFuses operations will be committed, including those from the ongoing SBv2 initialization, which might leave the chip in an inconsistent state.

We need to figure out a way to avoid this scenario.

Reply:
I've placed the Flash Encryption procedure after the Secure Boot, so the validations for SB and MCUboot would be already done and the SB eFuse operations would not interfere with the FE eFuse operations.

Reply:
Nice, thanks!